test/unit/lib/aws/awssignv4signer-versions.test.js

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ */
+const { test } = require('tap');
+
+const AwsSigv4Signer = require('../../../../lib/aws/AwsSigv4Signer');
+const AwsSigv4SignerV3 = require('../../../../lib/aws/AwsSigv4Signer-sdk-v3');
+const AwsSigv4SignerError = require('../../../../lib/aws/errors');
+
+const { Client, buildServer } = require('../../../utils');
+
+test('Sign with SigV4 using default getCredentials provider', (t) => {
+  t.plan(2);
+
+  function handler(req, res) {
+    res.setHeader('Content-Type', 'application/json;utf=8');
+    res.end(JSON.stringify({ hello: 'world' }));
+  }
+
+  buildServer(handler, ({ port }, server) => {
+    const mockRegion = 'us-east-1';
+
+    const AwsSigv4SignerOptions = {
+      region: mockRegion,
+    };
+
+    const client = new Client({
+      ...AwsSigv4Signer(AwsSigv4SignerOptions),
+      node: `http://localhost:${port}`,
+    });
+
+    client
+      .search({
+        index: 'test',
+        q: 'foo:bar',
+      })
+      .catch((err) => {
+        t.ok(err instanceof AwsSigv4SignerError);
+        t.same(
+          err.message,
+          'Unable to find a valid AWS SDK, please provide a valid getCredentials function to AwsSigv4Signer options.'
+        );
+      })
+      .finally(() => {
+        server.stop();
+      });
+  });
+});
+
+test('Sign with SigV4 using default getCredentials provider aws sdk v3', (t) => {
+  t.plan(2);
+
+  function handler(req, res) {
+    res.setHeader('Content-Type', 'application/json;utf=8');
+    res.end(JSON.stringify({ hello: 'world' }));
+  }
+
+  buildServer(handler, ({ port }, server) => {
+    const mockRegion = 'us-east-1';
+
+    const AwsSigv4SignerOptions = {
+      region: mockRegion,
+    };
+    const client = new Client({
+      ...AwsSigv4SignerV3(AwsSigv4SignerOptions),
+      node: `http://localhost:${port}`,
+    });
+
+    client
+      .search({
+        index: 'test',
+        q: 'foo:bar',
+      })
+      .catch((err) => {
+        t.ok(err instanceof AwsSigv4SignerError);
+        t.same(
+          err.message,
+          "Missing '@aws-sdk/credential-provider-node' module. Install it as a dependency."
+        );
+      })
+      .finally(() => {
+        server.stop();
+      });
+  });
+});
```

Comment:
Are these tests only testing if they throw the correct errors not for paths where no errors are thrown? I think that's why Codecov is not happy.

Reply:
This would require mocking the import as the dependencies are not installed. While tap supports mocking imports, the project does not use the latest tap version which has breaking changes. 

Doesn't seem like there is any other way to test dependency missing/present scenarios without mocking.

Reply:
We should solve this with integration tests IMO, which is a separate affair.

Reply:
I'm thinking of setting up different `package.json` for each test case like you have in [test/bundlers/esbuild-test/package.json](https://github.com/opensearch-project/opensearch-js/pull/758/files/b0eb72e76c8f17490dfb381664dd06d2cb9ec2fe#diff-1cb683c49678d4831e23c772e985f2d08b5af71d8c0d3cc931399df98768069c)

Reply:
It's a good idea. Do we want to hold this PR until then? Open an issue and do it later?